### PR TITLE
added the necessary flink-region flag to be able to grant flinkDevelo…

### PR DESCRIPTION
**Pull request by luca-filipponi**
For granting flinkDevelper on a single computePool, the crn need to include the region as well

Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- PLACEHOLDER

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok
   * no: DO NOT MERGE until the required features are live in prod  

What
----

References
----------

Test & Review
-------------

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/iam/command_rbac_role_binding_create.go [hunk 1]
```diff
@@ -49,6 +49,14 @@ func (c *roleBindingCommand) newCreateCommand() *cobra.Command {
 				Text: `Grant the "ResourceOwner" role to principal "User:u-123456" and subject "test" in schema context "schema_context" for Schema Registry "lsrc-123456" in the environment "env-12345":`,
 				Code: `confluent iam rbac role-binding create --principal User:u-123456 --role ResourceOwner --environment env-12345 --schema-registry-cluster lsrc-123456 --resource "Subject::.schema_context:test"`,
 			},
+			examples.Example{
+				Text: `Grant the role "FlinkDeveloper" to the principal "User:u-123456" in the environment "env-12345":`,
+				Code: "confluent iam rbac role-binding create --principal User:u-123456 --role FlinkDeveloper --environment env-12345",
+			},
+			examples.Example{
+				Text: `Grant the role "FlinkDeveloper" to the principal "User:u-123456", in the environment "env-12345" for the compute pool "lfcp-123456" in the Flink region "us-east-2":`,
+				Code: "confluent iam rbac role-binding create --principal User:u-123456 --role FlinkDeveloper --environment env-12345 --flink-region aws.us-east-2 --resource compute-pool:lfcp-123456",
```
Comment (keahung) [inline internal/iam/command_rbac_role_binding_create.go:58]: Can we update the `compute-pool` resource type to `ComputePool`? All the other resourcetypes seem to imply CamelCase (e.g. `Topic`, `Subject`, `TransactionId`)

Comment (luca-filipponi) [reply internal/iam/command_rbac_role_binding_create.go:58]: sure, will do

File: test/fixtures/output/iam/rbac/role-binding/create-help.golden [hunk 2]
```diff
@@ -32,11 +32,20 @@ Grant the "ResourceOwner" role to principal "User:u-123456" and subject "test" i
 
   $ confluent iam rbac role-binding create --principal User:u-123456 --role ResourceOwner --environment env-12345 --schema-registry-cluster lsrc-123456 --resource "Subject::.schema_context:test"
 
+Grant the role "FlinkDeveloper" to the principal "User:u-123456", in the environment "env-12345":
+
+  $ confluent iam rbac role-binding create --principal User:u-123456 --role FlinkDeveloper --environment env-12345
+
+Grant the role "FlinkDeveloper" to the principal "User:u-123456", in the environment "env-12345" for the ComputePool "lfcp-123456" in the Flink region "us-east-2":
+
+  $ confluent iam rbac role-binding create --principal User:u-123456 --role FlinkDeveloper --environment env-12345 --flink-region aws.us-east-2 --resource compute-pool:lfcp-123456
```
Comment (brianstrauch) [inline test/fixtures/output/iam/rbac/role-binding/create-help.golden:41]:
```suggestion
Grant the role "FlinkDeveloper" to the principal "User:u-123456", in the environment "env-12345" for the compute pool "lfcp-123456" in the Flink region "us-east-2":

  $ confluent iam rbac role-binding create --principal User:u-123456 --role FlinkDeveloper --environment env-12345 --flink-region aws.us-east-2 --resource ComputePool:lfcp-123456
```

File: test/fixtures/output/iam/rbac/role-binding/create-help.golden [hunk 3]
```diff
@@ -32,11 +32,20 @@ Grant the "ResourceOwner" role to principal "User:u-123456" and subject "test" i
...
+
 Flags:
       --role string                      REQUIRED: Role name of the new role binding.
       --principal string                 REQUIRED: Qualified principal name for the role binding.
       --environment string               Environment ID for scope of role-binding operation.
       --current-environment              Use current environment ID for scope.
+      --flink-region string              Flink region  ID for the role binding.
```
Comment (brianstrauch) [inline test/fixtures/output/iam/rbac/role-binding/create-help.golden:48]:
```suggestion
      --flink-region string              Flink region ID for the role binding.
```

**Review by brianstrauch — APPROVED** [review]

**Review by brianstrauch — CHANGES_REQUESTED** [review]
The `--flink-region` flag needs to be put behind a feature flag, right?

File: internal/iam/command_rbac_role_binding_create.go (hunk 1 above)
Comment (keahung) [reply internal/iam/command_rbac_role_binding_create.go:58]: It's not updated on my end?

Comment (luca-filipponi) [reply internal/iam/command_rbac_role_binding_create.go:58]: fixin this now

File: internal/iam/command_rbac_role_binding.go [hunk 4]
```diff
@@ -159,10 +161,13 @@ func (c *roleBindingCommand) parseCommon(cmd *cobra.Command) (*roleBindingOption
 /*
 Helper function to add flags for all the legal scopes/clusters for the command.
 */
-func addClusterFlags(cmd *cobra.Command, isCloudLogin bool, cliCommand *pcmd.CLICommand) {
-	if isCloudLogin {
+func addClusterFlags(cmd *cobra.Command, cfg *config.Config, cliCommand *pcmd.CLICommand, dynamicContext *dynamicconfig.DynamicContext) {
+	if cfg.IsCloudLogin() {
 		cmd.Flags().String("environment", "", "Environment ID for scope of role-binding operation.")
 		cmd.Flags().Bool("current-environment", false, "Use current environment ID for scope.")
+		if cfg.IsTest || featureflags.Manager.BoolVariation("cli.flink.open_preview", dynamicContext, config.CliLaunchDarklyClient, true, false) {
```
Comment (MuweiHe) [inline internal/iam/command_rbac_role_binding.go:168]: what's the plan of this feature flag? will it be removed after all features are released?

**Review by MuweiHe — APPROVED** [review]

**Review by keahung — APPROVED** [review]

File: internal/iam/command_rbac_role_binding.go [hunk 5]
```diff
@@ -3,6 +3,8 @@ package iam
 import (
 	"context"
 	"fmt"
+	dynamicconfig "github.com/confluentinc/cli/v3/pkg/dynamic-config"
+	"github.com/confluentinc/cli/v3/pkg/featureflags"
```
Comment (brianstrauch) [inline internal/iam/command_rbac_role_binding.go:7]: `make lint`

File: internal/iam/command_rbac_role_binding.go [hunk 6]
```diff
@@ -159,10 +161,13 @@ func (c *roleBindingCommand) parseCommon(cmd *cobra.Command) (*roleBindingOption
 /*
 Helper function to add flags for all the legal scopes/clusters for the command.
 */
-func addClusterFlags(cmd *cobra.Command, isCloudLogin bool, cliCommand *pcmd.CLICommand) {
-	if isCloudLogin {
+func addClusterFlags(cmd *cobra.Command, cfg *config.Config, cliCommand *pcmd.CLICommand, dynamicContext *dynamicconfig.DynamicContext) {
```
Comment (brianstrauch) [inline internal/iam/command_rbac_role_binding.go:164]:
```suggestion
func addClusterFlags(cmd *cobra.Command, cfg *config.Config, cliCommand *pcmd.CLICommand, ctx *dynamicconfig.DynamicContext) {
```

File: internal/iam/command_rbac_role_binding.go (hunk 4 above)
Comment (brianstrauch) [reply internal/iam/command_rbac_role_binding.go:168]: Yes, this'll be able to be removed after open preview

File: internal/iam/command_rbac_role_binding_create.go [hunk 7]
```diff
@@ -49,6 +49,14 @@ func (c *roleBindingCommand) newCreateCommand() *cobra.Command {
 				Text: `Grant the "ResourceOwner" role to principal "User:u-123456" and subject "test" in schema context "schema_context" for Schema Registry "lsrc-123456" in the environment "env-12345":`,
 				Code: `confluent iam rbac role-binding create --principal User:u-123456 --role ResourceOwner --environment env-12345 --schema-registry-cluster lsrc-123456 --resource "Subject::.schema_context:test"`,
 			},
+			examples.Example{
```
Comment (brianstrauch) [inline internal/iam/command_rbac_role_binding_create.go:52]: Should these also belong behind the feature flag?

File: internal/iam/command_rbac_role_binding.go (hunk 4 above)
Comment (luca-filipponi) [reply internal/iam/command_rbac_role_binding.go:168]: yea, we have an OP flink for flink that we are going to get rid as soon we release

File: internal/iam/command_rbac_role_binding.go (hunk 5 above)
Comment (luca-filipponi) [reply internal/iam/command_rbac_role_binding.go:7]: will do

File: internal/iam/command_rbac_role_binding_create.go (hunk 7 above)
Comment (luca-filipponi) [reply internal/iam/command_rbac_role_binding_create.go:52]: argh, yea

File: internal/iam/command_rbac_role_binding.go (hunk 6 above)
Comment (luca-filipponi) [reply internal/iam/command_rbac_role_binding.go:164]: done

File: internal/iam/command_rbac_role_binding_create.go [hunk 8]
```diff
@@ -19,8 +21,10 @@ func (c *roleBindingCommand) newCreateCommand() *cobra.Command {
 		RunE:  c.create,
 	}
 
+	var exs []examples.Example
```
Comment (brianstrauch) [inline internal/iam/command_rbac_role_binding_create.go:24]:
```suggestion
	var examples []examples.Example
```

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**